### ISSUE TITLE
It not possible to make 2 people with different Lora

### ISSUE THREAD

**Issue by levelmaxcat** [opparco/stable-diffusion-webui-two-shot#14]
I try Two shot extension  https://github.com/opparco/stable-diffusion-webui-two-shot

It not possible to make 2 people with different Lora

example.

2girls
AND 2girls, black hair , <lora:A>
AND 2girls, blonde hair , <lora:B>

result always be 2girls ,same face with <lora:A>+<lora:B>

**Comment by ClashSAN** [comment]
your actual desire might be to create 1 lora with the 2 characters from 2 loras with 1 character each.

Maybe a merging method already exists to do this effectively, for checkpoints?
Then you could extract your lora.

https://github.com/ogkalu2/Merge-Stable-Diffusion-models-without-distortion ?
I don't know if this works better for dreambooth models, I've only tried style models.

**Comment by scruffynerf** [comment]
I was totally able to use 2 lora in one image.

I split the space into 3 plus the entire space (so 4 sections)
I used a Barbara Eden lora in 3 out of the 4, and a Robot Lora in the middle of the 3... 

LORAs are only loaded once, so make sure LORAs are loaded with the prompt part using it.

Results included:

![00193-792545082-barbara_eden_AND_barbara_eden_AND_robot_AND_barbara_eden](https://user-images.githubusercontent.com/21144471/220005499-c46e3461-02f1-472a-a7f7-303d0e934e50.png)
![00191-792545080-barbara_eden_AND_barbara_eden_AND_robot_AND_barbara_eden](https://user-images.githubusercontent.com/21144471/220005653-de35bc1e-1150-4f97-b2c9-56d1f3ca4719.png)
![00188-3447715649-a_pretty_woman_AND_barbara_eden_AND_robot_AND_barbara_eden](https://user-images.githubusercontent.com/21144471/220005693-3d547b9a-8e9b-4d6d-9d66-7cc09dc54d7d.png)
![00194-792545083-barbara_eden_AND_barbara_eden_AND_robot_AND_barbara_eden](https://user-images.githubusercontent.com/21144471/220005716-f1f463e0-e62a-44d3-9fe1-5f1346a22e9d.png)
![00192-792545081-barbara_eden_AND_barbara_eden_AND_robot_AND_barbara_eden](https://user-images.githubusercontent.com/21144471/220005743-38685fc4-4437-4051-b144-741a851bee6c.png)

If you were correct, she'd be entirely robotic, but the outside edges are normal (Eden Lora) and only the middle is Robot (Robot Lora) (Cover the rest of the photo in 3rds, and look.)

**Comment by cloudman123** [comment]
> I try Two shot extension https://github.com/opparco/stable-diffusion-webui-two-shot
> 
> It not possible to make 2 people with different Lora
> 
> example.
> 
> 2girls AND 2girls, black hair , lora:A AND 2girls, blonde hair , lora:B
> 
> result always be 2girls ,same face with lora:A+lora:B

same issue
maybe not support?

**Comment by opparco** [comment]
webui issue, not this extension.
extra_networks.parse_prompt does not consider the position on the prompt (i.e. which subprompt to apply to) when reading lora.

**Comment by scruffynerf** [comment]
I got it to work, by making sure the triggers of the lora were only in the prompt parts I wanted:

it's a tricky balance but doable:
prompt: `2girls AND <lora:1xsheHulk_:.5> shehulk wearing a shirt AND 2girls <lora:LuisapSuperCopyright_v1:.8> red cap`
standard default 3 part split, (whole plus 2 halfs), and did weights of 0.5,1,0.4

took a few tries but...

![00071-840416034-2girls AND  shehulk wearing a shirt AND 2girls  red cap](https://user-images.githubusercontent.com/21144471/220012032-f73a26dc-7984-4690-949e-a02037500b3d.png)

![00064-1658389063-2girls AND  shehulk wearing a shirt AND 2girls    red cap](https://user-images.githubusercontent.com/21144471/220012077-46ffc71e-8b24-4729-9eaf-96187c64ff7c.png)

![00066-757020145-2girls AND  shehulk wearing a shirt AND 2girls  red cap](https://user-images.githubusercontent.com/21144471/220012107-a540f4d6-da72-4ac6-b9c0-52897010a77d.png)

you need to adjust loras to not bleed thru, use triggered loras, and weight to reduce the overlap. (again, cover half the image and notice the dividing line...)

**Comment by levelmaxcat** [comment]
try this prompt

`2girls AND 2girls <lora:1xsheHulk_:.5> wearing blue cap AND 2girls <lora:LuisapSuperCopyright_v1:.8> wearing red cap`


this works or not?

**Comment by scruffynerf** [comment]
> 2girls AND 2girls <lora:1xsheHulk_:.5> wearing blue cap AND 2girls <lora:LuisapSuperCopyright_v1:.8> wearing red cap

the above lack the lora trigger 'shehulk', so the cap pulls in the other lora, and they look more alike.
adding shehulk fixes that... and you can see the cap triggers the other lora, so it's similar...
going with a different word/color like 'purple hat' tended (but not always) to avoid the similar hat


examples:
![01855-1446048692-2girls AND 2girls  shehulk with a blue cap  AND 2girls  wearing red cap](https://user-images.githubusercontent.com/21144471/220563048-9517fa37-e4bd-4b81-a4ce-9696de5058c8.png)

![01837-2359549834-2girls AND 2girls  wearing blue cap AND 2girls  wearing red cap](https://user-images.githubusercontent.com/21144471/220562979-8dff6c10-aa20-49c1-9227-ca953d885ebd.png)

![01842-3751822815-2girls AND 2girls  shehulk with a purple hat  AND 2girls  wearing red cap](https://user-images.githubusercontent.com/21144471/220563647-65818231-d83e-4c90-aa74-8405ec2e1082.png)


In summary, it's doable, it's not impossible.  You have to be aware of the potential for bleed, use triggered loras (and not lora that just add influence), and try a bit.  As mentioned, there is definitely a technical issue with the parsing, so it's not truly isolated.  But it's very doable.  I suggest closing this ticket.

**Comment by Padusarbor** [comment]
https://github.com/opparco/stable-diffusion-webui-composable-lora
